### ISSUE TITLE
data.aws_organizations_organization doesn't pull all accounts when used from child account

### ISSUE THREAD

**Issue by rnikoopour** [hashicorp/terraform-provider-aws#20899]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.15.3
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.58.0

### Affected Resource(s)

* data.aws_organizations_organization

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
terraform {}
provider "aws" {region = "us-west-2"}
data "aws_organizations_organization" "org" {}

output "org" {
  value = data.aws_organizations_organization.org
}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior

When using a child account to run the terraform I expect to see all my organizations accounts in `data.aws_orgnizations_organization.org.accounts` just like I do when I call `aws organizations list-accounts`. 

### Actual Behavior

`data.aws_orgnizations_organization.org.accounts` is null

### Steps to Reproduce

1. Log into child account cia the CLI
1. `terraform apply` the above hcl

### Important Factoids

Adding to the account fields in blocked by requiring the terraform to be running as the root account: 
https://github.com/hashicorp/terraform-provider-aws/blob/118137cad03ab6521b5de6e44aaa8f35b89b7b0a/aws/data_source_aws_organizations_organization.go#L159

### References

N/A

**Comment by cob16** [comment]
This looks to be a duplicate of https://github.com/hashicorp/terraform-provider-aws/issues/18590
